models/User.model.js

```diff
@@ -0,0 +1,59 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcryptjs');
+
+const Schema = mongoose.Schema;
+
+const userSchema = new Schema({
+    firstName:{
+        type: String,
+        required: [true, 'Please enter your first name']
+    },
+
+    lastName:{
+        type: String,
+        required: [true, 'Please enter your last name']
+    },
+
+
+    email:{
+        type: String,
+        required: true,
+        unique: true,
+        match: [/^\S+@\S+\.\S+$/, 'Please enter a valid email address']
+    },
+
+    password:{
+        type: String,
+        required: [true, 'Please enter a password'],
+        minlength: [6, 'Your password must be at least 8 characters long']
+    }
+}, {
+    timestamps: true
+});
+
+
+//encrypt password before saving
+userSchema.pre('save', async function(next){
+    if(!this.isModified('password'))
+        next();
+    try{
+        const salt = await bcrypt.genSalt(10);
```

Comment:
afair, since we required bcryptjs, we should use genSaltSync, hashSync and compareSync. correct me if I'm wrong please

Reply:
No worries, I'm using the Asynchronous method. Reason: to allow Node.js to handle other tasks while the bcrypt operations are running(multitasking)...but no worries, will change it to Synchronous one

Reply:
thanks for the memory refresh! it's all coming back to me now